### ISSUE TITLE
Python3: arraypad.py's "_normalize_shape" not accepting all types of ints

### ISSUE THREAD

**Issue by fmaussion** [numpy/numpy#5499]
This is Numpy V1.9.1. I found the bug while testing a working py2 code in py3

```python
np.pad(np.zeros((3,3)), np.int(3), 'symmetric')
```
is working fine, but:

```python
np.pad(np.zeros((3,3)), np.int64(3), 'symmetric')
```
is crashing with:

```python
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/mowglie/.pyvirtualenvs/py3.3/lib/python3.3/site-packages/numpy/lib/arraypad.py", line 1292, in pad
    pad_width = _validate_lengths(narray, pad_width)
  File "/home/mowglie/.pyvirtualenvs/py3.3/lib/python3.3/site-packages/numpy/lib/arraypad.py", line 1079, in _validate_lengths
    normshp = _normalize_shape(narray, number_elements)
  File "/home/mowglie/.pyvirtualenvs/py3.3/lib/python3.3/site-packages/numpy/lib/arraypad.py", line 1044, in _normalize_shape
    raise ValueError(fmt % (shape,))
ValueError: Unable to create correctly shaped tuple from 3
```

it's related to the fact that ```isinstance(np.int64(3), int)``` returns ```False``` in Python3 but ```True``` in Python2. 

Replacing ```isinstance(shape, int)``` with ```isinstance(shape, (int, np.integer))``` in arraypad.py at line 1024 should do the trick. 
I could do the fix but I never commited a thing before....

**Comment by jan-glx** [comment]
I ran into problems with this as well, but it is probably already fixed in 29464a96941b61b91d703dc20c3f426466df0c2a. @JDWarner, you may want to close this.

**Comment by jaimefrio** [comment]
Do you have a way of testing if the error persists in current master? It does seems like the latest `np.pad` handles things differently and shouldn't have that problem.

The issue with `np.int64` not being an `int` is probably related to #5801.

**Comment by JDWarner** [comment]
@jan-glx thanks for the ping. You're correct, this was fixed in 29464a9 when I overhauled the type-checking infrastructure behind `np.pad`. The current version converts `pad_width` into a NumPy array and works from there, checking `arr.dtype.kind == 'i'` which should be correct regardless of Python version as it avoids the `isinstance` check.

@jaimefrio The relevant test would be 

``` python
import numpy as np
from numpy.lib.arraypad import _normalize_shape
from numpy.testing import assert_array_equal

orig_arr = np.zeros((3, 4))

pad_amt0 = np.int(3)
pad_amt1 = np.int64(3)
expected = ((3, 3), (3, 3))

assert_array_equal(expected, _normalize_shape(orig_arr, pad_amt0))
assert_array_equal(expected, _normalize_shape(orig_arr, pad_amt1))
```

which I could add to the suite if you like.

**Comment by JDWarner** [comment]
I can't close this issue personally, but I have now verified the above issue does in fact fail as advertised in 1.9.2 under Python 3.x but is fixed in the master branch since #5361 was merged.

**Comment by njsmith** [comment]
Ok, thanks!
